packages/connect-node-conformance/src/invoke.ts

```diff
@@ -0,0 +1,288 @@
+    resHeaders = convertToProtoHeaders(error.metadata);
+    resTrailers = convertToProtoHeaders(error.metadata);
+  }
+  return new ClientResponseResult({
+    payloads: payloads,
+    responseHeaders: resHeaders,
+    responseTrailers: resTrailers,
+    error: convertToProtoError(error),
+  });
+}
+
+async function serverStream(
+  client: ConformanceClient,
+  req: ClientCompatRequest,
+) {
+  if (req.requestMessages.length !== 1) {
+    throw new Error("ServerStream method requires exactly one request message");
+  }
+  const msg = req.requestMessages[0];
+  const uReq = new ServerStreamRequest();
+  if (!msg.unpackTo(uReq)) {
+    throw new Error(
+      "Could not unpack request message to server stream request",
+    );
+  }
+  const reqHeader = new Headers();
+  appendProtoHeaders(reqHeader, req.requestHeaders);
+  let error: ConnectError | undefined = undefined;
+  let resHeaders: ConformanceHeader[] = [];
+  let resTrailers: ConformanceHeader[] = [];
+  const payloads: ConformancePayload[] = [];
+  try {
+    for await (const msg of client.serverStream(uReq, {
+      headers: reqHeader,
+      onHeader(headers) {
+        resHeaders = convertToProtoHeaders(headers);
+      },
+      onTrailer(trailers) {
+        resTrailers = convertToProtoHeaders(trailers);
+      },
+    })) {
+      payloads.push(msg.payload!);
+    }
+  } catch (e) {
+    error = ConnectError.from(e);
+    resHeaders = convertToProtoHeaders(error.metadata);
+    resTrailers = convertToProtoHeaders(error.metadata);
+  }
+  return new ClientResponseResult({
+    responseHeaders: resHeaders,
+    responseTrailers: resTrailers,
+    payloads: payloads,
+    error: convertToProtoError(error),
+  });
+}
+
+async function clientStream(
+  client: ConformanceClient,
+  req: ClientCompatRequest,
+) {
+  const reqHeaders = new Headers();
+  appendProtoHeaders(reqHeaders, req.requestHeaders);
+  let error: ConnectError | undefined = undefined;
+  let resHeaders: ConformanceHeader[] = [];
+  let resTrailers: ConformanceHeader[] = [];
+  const payloads: ConformancePayload[] = [];
+  try {
+    const csRes = await client.clientStream(
+      (async function* () {
+        for (const msg of req.requestMessages) {
+          const csReq = new ClientStreamRequest();
+          if (!msg.unpackTo(csReq)) {
+            throw new Error(
+              "Could not unpack request message to client stream request",
+            );
+          }
+          await wait(req.requestDelayMs);
+          yield csReq;
+        }
+      })(),
+      {
+        headers: reqHeaders,
+        onHeader(headers) {
+          resHeaders = convertToProtoHeaders(headers);
+        },
+        onTrailer(trailers) {
+          resTrailers = convertToProtoHeaders(trailers);
+        },
+      },
+    );
+    payloads.push(csRes.payload!);
+  } catch (e) {
+    error = ConnectError.from(e);
+    resHeaders = convertToProtoHeaders(error.metadata);
+    resTrailers = convertToProtoHeaders(error.metadata);
```

Comment:
There reference client does not use error metadata at all for streaming: https://github.com/connectrpc/conformance/blob/34f2af36604771b02138b70db3f0f4502b8d957b/internal/app/referenceclient/impl.go#L157-L162

It's good that we provide combined headers _and_ trailers in error metadata, but it doesn't feel right to populate both `ClientResponseResult.response_headers` and `ClientResponseResult.response_trailers` with them.

If our clients don't invoke the `onHeader` and `onTrailer` callbacks on error, I think we should change them to do so.

Reply:
With interceptors, I am not sure how we can do that. 

The way I see it is that the promise clients are just one flavor of API, we also have the callback flavor, but should we be testing them as part of conformance? We can just as easily make a client that is close to how requests and responses are done in go and use that for conformance or better yet just use the transport directly.

Reply:
My bad, I didn't have the `Transport` types correctly in mind. We can't invoke onTrailer callbacks on error. We should not start differentiating in `ConnectError`. 

It still doesn't feel right to populate both `ClientResponseResult.response_headers` and `ClientResponseResult.response_trailers` with combined metadata. We need at least an explanatory comment.

Reply:
Added the comment and also switched to only adding the header, if not already populated.